src/main/java/build/buf/protovalidate/internal/celext/CustomOverload.java

```diff
@@ -94,29 +104,14 @@ private static Overload binaryFormat() {
    *
    * @return The {@link Overload} instance for the "unique" operation.
    */
-  private static Overload unaryUnique() {
+  private static Overload unique() {
     return Overload.unary(
-        "unique",
+        OVERLOAD_UNIQUE,
         (val) -> {
-          switch (val.type().typeEnum()) {
-            case List:
-              Lister lister = (Lister) val;
-              if (lister.size().intValue() == 0L) {
-                // Uniqueness for empty lists are true.
-                return BoolT.True;
-              }
-              Val firstValue = lister.get(IntT.intOf(0));
-              return unaryOpForPrimitiveVal(firstValue).invoke(lister);
-            case Bool:
-            case Bytes:
-            case Double:
-            case Int:
-            case String:
-            case Uint:
-              return unaryOpForPrimitiveVal(val).invoke(val);
```

Comment:
A lot of these weren't valid for unique() - it should only operate on list types according to the Go implementation.